src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -148,35 +158,106 @@ public void visitJavadocToken(DetailNode ast) {
     }
 
     /**
-     * Finds if inheritDoc is placed properly in java doc.
+     * Checks if period is at the end of sentence.
      * @param ast Javadoc root node.
-     * @return true if inheritDoc is valid or false.
+     * @return error string
      */
-    private static boolean isOnlyInheritDoc(DetailNode ast) {
-        boolean extraTextFound = false;
-        boolean containsInheritDoc = false;
+    private String checkPeriodAtEnd(DetailNode ast) {
```

Comment:
all checks should be done in `visitJavadocToken` method, all you need is to have smart method that return String of first sentence from javadoc content.